emulated-test/test/frequency-westend/src/tests/barrier_tests.rs

```diff
@@ -0,0 +1,103 @@
+use crate::imports::Barrier;
+use frame_support::weights::Weight;
+use staging_xcm::prelude::*;
+use xcm_executor::traits::{Properties, ShouldExecute as XcmBarrier};
+
+#[test]
+fn test_barrier_weight_credit_should_pass() {
+	let location = Location::new(100, []);
+	let mut instructions =
+		Xcm::<()>(vec![TransferAsset { assets: (Parent, 100).into(), beneficiary: Here.into() }]);
+	let mut properties =
+		Properties { weight_credit: Weight::from_parts(1_000_000_000, 0), message_id: None };
+	let weight = Weight::from_parts(1_000_000_000, 0);
+
+	let result = <Barrier as XcmBarrier>::should_execute(
+		&location,
+		instructions.inner_mut(),
+		weight,
+		&mut properties,
+	);
+
+	assert!(result.is_ok(), "Barrier test failed: {:?}", result);
+}
+
+#[test]
+fn test_barrier_weight_credit_should_fail() {
```

Comment:
It's not clear from this test name why this should fail and the above one should pass.  Please name this something more descriptive so we can know what you are trying to test.